envi/cli.py

```diff
@@ -476,6 +478,17 @@ def do_script(self, line):
         locals = self.getExpressionLocals()
         locals['argv'] = argv
 
+        if len(argv) and argv[0] == "?":
+            scripts = []
+            for scriptdir in self.scriptpaths:
+                # FIXME: filter on more than just ".py".  something internal
+                pscripts = [py for py in os.listdir(scriptdir) if self.validate_script(scriptdir + os.sep + py)]
```

Comment:
Since the command's actual use would allow script foo/bar/baz.py we would probably also want to consider something more recursive like walk() to list scripts within nested directories.